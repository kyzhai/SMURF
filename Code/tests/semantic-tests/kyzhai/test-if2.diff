1c1,10
< Fatal error: exception Sast.Main_missing
---
> Program: 
> 	SVardef: 
> 		ID: a :: Bool
> 
> 	if true then true else false
> Global Scope: 
> 	ID: a :: Bool
> 	ID: print :: Unknown
> 	ID: random :: Unknown
> 
