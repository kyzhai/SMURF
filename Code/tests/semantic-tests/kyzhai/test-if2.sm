a = if True then True else False

main = []
