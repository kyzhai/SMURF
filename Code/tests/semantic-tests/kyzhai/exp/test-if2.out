Program: 
	SVardef: 
		ID: a  :: Bool

	if true then true else false
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: a  :: Bool
	ID: print x :: a -> a
	ID: random  :: Int

